### PR TITLE
Reject ReDoS-prone Muon exclude_layers to avoid catastrophic backtracking

### PR DESCRIPTION
**Pull request by LinZiyuu**
## Description

`Muon._should_use_adamw` decides, per variable, whether to fall back to the
AdamW update by matching the variable's path against each entry of
`exclude_layers` with `re.search`:

```python
for keyword in self.exclude_layers:
    if re.search(keyword, variable.path):
        return True
```

`exclude_layers` is an ordinary constructor argument that `get_config`
serializes, so a malicious model carries it. An entry crafted for catastrophic
backtracking (e.g. `(a+)+$`) pins a CPU core for an unbounded amount of time on
the first gradient step after the model is loaded with `compile=True` and
trained — a ReDoS (CWE-1333). No special flags are required; the regex runs on
the default training path.

The entries are documented as plain layer-name keywords ("keywords of layer
names to exclude", e.g. `exclude_layers=["output_dense"]`), so this PR validates
each entry at construction — and therefore at deserialization, before any
training step — against an allowlist: an entry must be a run of path characters,
optionally ending in `.*` (or be `.*` itself). This admits every legitimate
keyword while making any backtracking-prone construct (nested quantifiers,
alternation groups, multiple wildcards) impossible. The validation regex is
itself linear (one quantifier over a character class), so it cannot be turned
into a ReDoS.

This mirrors the `DTypePolicyMap` key guard in #23102, which addresses the same
config-controlled-regex ReDoS class on the model-loading path. A malicious model
now fails to load with a clear error instead of hanging the trainer.

Adds a regression test: backtracking-prone and over-long entries are rejected,
while plain keywords and a simple trailing `.*` glob are still accepted.

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/optimizers/muon.py [hunk 1]
```diff
@@ -4,6 +4,42 @@
...
+# and the list is a constructor argument serialized into a model's config, so a
+# crafted entry can cause catastrophic regex backtracking (ReDoS, CWE-1333) that
+# hangs the first training step after an untrusted model is loaded. The entries
+# are documented as plain layer-name keywords (e.g. `["output_dense"]`), so we
+# restrict each one to a run of path characters, optionally ending in `.*` (or
+# `.*` itself). This admits every legitimate keyword while making any
+# backtracking-prone construct impossible; the validation regex is itself linear
+# (one quantifier over a character class), so it cannot be turned into a ReDoS.
+_MAX_EXCLUDE_LAYER_LENGTH = 256
+_SAFE_EXCLUDE_LAYER_RE = re.compile(r"^(?:[a-zA-Z0-9_./-]+(?:\.\*)?|\.\*)$")
```
Comment (hertschuh) [inline keras/src/optimizers/muon.py:17]: Same comment as https://github.com/keras-team/keras/pull/23102#discussion_r3599359526

**Review by hertschuh — CHANGES_REQUESTED** [review]
